### DISCUSSION TITLE
Getting Filenames of files from a Google Cloud Storage

### DISCUSSION THREAD

**Discussion by mark-cale** [redpanda-data/connect#2519] [Q&A] [answered]
Hello,
I'm attempting to retrieve the filenames of files stored in my Google Cloud Storage. How can I achieve this? I've been experimenting with metadata, but so far, I haven't had any success. Any guidance would be greatly appreciated.

**Comment by mihaitodor** [comment, marked as answer]
Hey @mark-cale 👋 If you only need the filenames, the [`gcp_cloud_storage` input](https://www.benthos.dev/docs/components/inputs/gcp_cloud_storage) is going to be wasteful, since it will also read the contents of the files. If you also need the contents, then what I think you want is to look at the `gcs_key` metadata field. Try adding a log processor like so:

```yaml
- log:
     message: |
       Filename: ${! @gcs_key }
```

You can also dump all the metadata fields if you just use `@` instead of `@gcs_key`.

**Reply by mark-cale** [reply]
Cool Thanks!
